custom/icds_reports/tests/test_aggregation_script.py

```diff
@@ -117,298 +121,280 @@ def _fasterAssertListEqual(self, list1, list2):
+            sort_key=['state_id', 'district_id', 'block_id', 'supervisor_id', 'awc_id']
         )
 
     def test_agg_awc_2017_05_01_1(self):
         self._load_and_compare_data(
             'agg_awc_2017-05-01_1',
-            os.path.join(OUTPUT_PATH, 'agg_awc_2017-05-01_1.csv'),
-            lambda x: (x['state_id'], x['district_id'], x['block_id'], x['supervisor_id'], x['awc_id'])
+            os.path.join(OUTPUT_PATH, 'agg_awc_2017-05-01_1_sorted.csv'),
+            sort_key=['state_id', 'district_id', 'block_id', 'supervisor_id', 'awc_id']
         )
 
     def test_agg_awc_2017_05_01_2(self):
         self._load_and_compare_data(
             'agg_awc_2017-05-01_2',
-            os.path.join(OUTPUT_PATH, 'agg_awc_2017-05-01_2.csv'),
-            lambda x: (x['state_id'], x['district_id'], x['block_id'], x['supervisor_id'], x['awc_id'])
+            os.path.join(OUTPUT_PATH, 'agg_awc_2017-05-01_2_sorted.csv'),
+            sort_key=['state_id', 'district_id', 'block_id', 'supervisor_id', 'awc_id']
         )
 
     def test_agg_awc_2017_05_01_3(self):
         self._load_and_compare_data(
             'agg_awc_2017-05-01_3',
-            os.path.join(OUTPUT_PATH, 'agg_awc_2017-05-01_3.csv'),
-            lambda x: (x['state_id'], x['district_id'], x['block_id'], x['supervisor_id'], x['awc_id'])
+            os.path.join(OUTPUT_PATH, 'agg_awc_2017-05-01_3_sorted.csv'),
+            sort_key=['state_id', 'district_id', 'block_id', 'supervisor_id', 'awc_id']
         )
 
     def test_agg_awc_2017_05_01_4(self):
         self._load_and_compare_data(
             'agg_awc_2017-05-01_4',
-            os.path.join(OUTPUT_PATH, 'agg_awc_2017-05-01_4.csv'),
-            lambda x: (x['state_id'], x['district_id'], x['block_id'], x['supervisor_id'], x['awc_id'])
+            os.path.join(OUTPUT_PATH, 'agg_awc_2017-05-01_4_sorted.csv'),
+            sort_key=['state_id', 'district_id', 'block_id', 'supervisor_id', 'awc_id']
         )
 
     def test_agg_awc_2017_05_01_5(self):
         self._load_and_compare_data(
             'agg_awc_2017-05-01_5',
-            os.path.join(OUTPUT_PATH, 'agg_awc_2017-05-01_5.csv'),
-            lambda x: (x['state_id'], x['district_id'], x['block_id'], x['supervisor_id'], x['awc_id'])
+            os.path.join(OUTPUT_PATH, 'agg_awc_2017-05-01_5_sorted.csv'),
+            sort_key=['state_id', 'district_id', 'block_id', 'supervisor_id', 'awc_id']
         )
 
     def test_agg_child_health_2017_04_01_1(self):
         self._load_and_compare_data(
             'agg_child_health_2017-04-01_1',
-            os.path.join(OUTPUT_PATH, 'agg_child_health_2017-04-01_1.csv'),
-            lambda x: (
-                x['state_id'], x['district_id'], x['block_id'], x['supervisor_id'], x['awc_id'],
-                x['gender'], x['age_tranche'], x['caste'], x['disabled'], x['minority'], x['resident']
-            )
+            os.path.join(OUTPUT_PATH, 'agg_child_health_2017-04-01_1_sorted.csv'),
```

Comment:
Looks like these are failing in the tests. I think you'll have to sort by all the columns in the previous lambda instead of just `awc_id`

Reply:
The only failed test is `test_agg_awc_daily` (not this one). Even that didn't fail locally, I will fix it.

Reply:
For this case `awc_id` is enough for sorting the data

Reply:
This failure was super difficult to debug as it was not possible to reproduce locally. The sort result from DB depends on collation, so I had to use the same collation that the docker image uses to presort the CSV to fix the issue